bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/util/UnitUtils.java

```diff
@@ -82,6 +92,39 @@
         }
     }
 
+    /**
+     * The name of the dimension as stated in the ChannelType configuration.
+     * e.g.
+     * <p>
+     * <code> Unit: 'm' -> "Length"</code>
+     * <p>
+     * <code> Unit: 'kWh' -> "Energy"</code>
+     * <p>
+     * If the {@link Unit} can not be found in any of the available Measurement systems, it returns <code>null</code>
+     *
+     * @param unit The {@link Unit} to get the Dimension's name from.
+     * @return The Dimension string or null if the unit can not be found in any of the SystemOfUnits.
+     */
+    public static @Nullable String getDimensionName(Unit<?> unit) {
+        for (Class<? extends SystemOfUnits> system : ALL_SYSTEM_OF_UNITS) {
+            for (Field field : system.getDeclaredFields()) {
+                if (field.getType().isAssignableFrom(Unit.class) && Modifier.isStatic(field.getModifiers())) {
+                    try {
+                        String dimension = ((Class<?>) ((ParameterizedType) field.getGenericType())
+                                .getActualTypeArguments()[0]).getSimpleName();
+                        Unit<?> systemUnit = (Unit<?>) field.get(null);
+                        if (systemUnit.isCompatible(unit)) {
+                            return dimension;
+                        }
+                    } catch (Exception e) {
```

Comment:
Are you able to catch something more specific then `Exception`?

Reply:
There are many exceptions that can be thrown here that are defined in the used java methods. Catching all of them would be superfluous imho as nothing can be done anyhow apart from logging.
Though there are some cases which I can handle specifically (but also just log them with a specific log message):
- Unit constant has no generic type argument
- The Unit value is null

Reply:
Yes, please try to catch the specific exceptions. It's up to you wether you also want to do specific log entries, I would prefer it though.

Reply:
Ok now?